tests/events/client/test_events_client.py

```diff
@@ -1,21 +1,97 @@
+from typing import Type
 from unittest import mock
 
 import pytest
 from websockets.exceptions import ConnectionClosed
 
 from prefect.client.base import PrefectHttpxClient
-from prefect.events import Event
+from prefect.events import Event, get_events_client
 from prefect.events.clients import (
     PrefectCloudEventsClient,
     PrefectEphemeralEventsClient,
     PrefectEventsClient,
 )
-from prefect.settings import PREFECT_EXPERIMENTAL_EVENTS, temporary_settings
+from prefect.settings import (
+    PREFECT_API_KEY,
+    PREFECT_API_URL,
+    PREFECT_CLOUD_API_URL,
+    PREFECT_EXPERIMENTAL_EVENTS,
+    temporary_settings,
+)
 from prefect.testing.fixtures import Puppeteer, Recorder
 
 
+@pytest.fixture
```

Comment:
@desertaxle I added tests for all the variations and incorporated the new ephemeral client too